### PR TITLE
Isolation CITK 80X (rebased)

### PR DESCRIPTION
**Pull request by ishvetso**
update for CITK isolation module in 76X: included puppi-based isolation, map based veto (electron and photon) and cone veto (photons)

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

File: PhysicsTools/IsolationAlgos/plugins/BuildFile.xml [hunk 1]
```diff
@@ -6,5 +6,6 @@
   <use   name="DataFormats/PatCandidates"/>
   <use   name="DataFormats/TauReco"/>
   <use   name="DataFormats/TrackReco"/>
+  <use   name="DataFormats/PatCandidates"/>
```
Comment (slava77) [inline PhysicsTools/IsolationAlgos/plugins/BuildFile.xml]: please remove this line. It is a duplicate in 80X version.

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
another (less) broken IB, sigh.

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/13391	b45d01a
- code changes are OK, the same as in 74X version. All new code for use in analysis.
- jenkins tests pass where needed (compilation and code quality checks)

**Comment by davidlange6** [comment]
this time from code that passed all checks.. backing it out

**Comment by ishvetso** [comment]
This branch contains modules to run isolation sums for electrons and photons: with map based veto and cone based veto, all these types of isolations can done for done for PUPPI-based isolation.

Modules can run either on miniAOD or AOD: for miniAOD one can use PUPPI weights that are stored in the packedCandidate directly by using CITKPFIsolationSumProducerForPUPPI with empty string for puppiValueMap. One can also rerun PUPPI and use PUPPI collection as input: for cone-based veto one should specify name of PUPPI collection and can use CITKPFIsolationSumProducer, for map-based veto one should specify the name of PUPPI value map (which is the same name as the name of PUPPI collection) but the name of collection should be either packedPFCandidates or pfNoPileupCanidates depending whether one runs on miniAOD or AOD (this is also illustrated in example in python/). For CITK twiki please refer here:

https://twiki.cern.ch/twiki/bin/viewauth/CMS/CommonIDAndIsolationFW

Studies have been for PUPPI-based isolation have been presented here:

https://indico.cern.ch/event/494563/contribution/2/attachments/1225182/1793161/iPUPPI_photons_EGM_9February.pdf

https://indico.cern.ch/event/482699/contribution/2/attachments/1215812/1775490/Shvetsov_update_iPUPPI_electrons.pdf

Electron cone based veto is not touched in this PR: I will create another PR to update as it does follow pfIsolationVariables 100%.

**Pull request merged**